--- a/nvidia-settings-@PV@/src/Makefile
+++ b/nvidia-settings-@PV@/src/Makefile
@@ -37,14 +37,6 @@
 
 PKG_CONFIG            ?= pkg-config
 
-ifndef GTK2_CFLAGS
-  GTK2_CFLAGS          := $(shell $(PKG_CONFIG) --cflags gtk+-2.0)
-endif
-
-ifndef GTK2_LDFLAGS
-  GTK2_LDFLAGS         := -Wl,--no-as-needed $(shell $(PKG_CONFIG) --libs gtk+-2.0)
-endif
-
 ifndef GTK3_AVAILABLE
   GTK3_AVAILABLE := $(shell $(PKG_CONFIG) --exists gtk+-3.0 && echo 1)
 endif
@@ -117,11 +109,6 @@
 
 NVIDIA_SETTINGS_VERSION := $(NVIDIA_VERSION)
 
-GTK2LIB_NAME = libnvidia-gtk2.so
-GTK2LIB_DIR  = $(OUTPUTDIR)/gtk2
-GTK2LIB = $(OUTPUTDIR)/$(GTK2LIB_NAME)
-GTK2LIB_SONAME = $(GTK2LIB_NAME).$(NVIDIA_SETTINGS_VERSION)
-
 ifdef BUILD_GTK3LIB
   GTK3LIB_NAME = libnvidia-gtk3.so
   GTK3LIB_DIR  = $(OUTPUTDIR)/gtk3
@@ -185,7 +172,6 @@
 # for "--as-needed" linker behavior).
 LIBS += -lX11 -lXext -lm $(LIBDL_LIBS)
 
-GTK2_LIBS += $(GTK2_LDFLAGS)
 GTK3_LIBS += $(GTK3_LDFLAGS)
 
 # Include all the source lists; dist-files.mk will define NVIDIA_SETTINGS_SRC
@@ -204,7 +190,6 @@
 OBJS        = $(call BUILD_OBJECT_LIST,$(SRC))
 XCP_OBJS    = $(call BUILD_OBJECT_LIST,$(XCP_SRC))
 
-GTK2_OBJS    = $(call BUILD_OBJECT_LIST_WITH_DIR,$(GTK_SRC),$(GTK2LIB_DIR))
 GTK3_OBJS    = $(call BUILD_OBJECT_LIST_WITH_DIR,$(GTK_SRC),$(GTK3LIB_DIR))
 IMAGE_OBJS    = $(addprefix $(OUTPUTDIR)/,$(addsuffix .o,$(notdir $(IMAGE_FILES))))
 IMAGE_HEADERS = $(addprefix $(OUTPUTDIR)/,$(addsuffix .h,$(notdir $(IMAGE_FILES))))
@@ -229,9 +214,6 @@
 
 $(call BUILD_OBJECT_LIST,$(XCP_SRC)): CFLAGS += -fPIC
 
-$(call BUILD_OBJECT_LIST_WITH_DIR,$(GTK_SRC),$(GTK2LIB_DIR)): \
-    CFLAGS += $(GTK2_CFLAGS) -fPIC -I $(XCONFIG_PARSER_DIR)/..
-
 ifdef BUILD_GTK3LIB
   $(call BUILD_OBJECT_LIST_WITH_DIR,$(GTK_SRC),$(GTK3LIB_DIR)): \
       CFLAGS += $(GTK3_CFLAGS) -fPIC -I $(XCONFIG_PARSER_DIR)/..
@@ -255,14 +237,12 @@
 
 .PHONY: all install NVIDIA_SETTINGS_install clean clobber
 
-all: $(NVIDIA_SETTINGS) $(GTK2LIB) $(GTK3LIB)
+all: $(NVIDIA_SETTINGS) $(GTK3LIB)
 
 install: NVIDIA_SETTINGS_install NVIDIA_GTKLIB_install
 
-NVIDIA_GTKLIB_install: $(GTK2LIB) $(GTK3LIB)
+NVIDIA_GTKLIB_install: $(GTK3LIB)
 	$(MKDIR) $(LIBDIR)
-	$(INSTALL) $(INSTALL_LIB_ARGS) $(GTK2LIB) \
-	    $(LIBDIR)/$(GTK2LIB_SONAME)
 ifdef BUILD_GTK3LIB
 	$(INSTALL) $(INSTALL_LIB_ARGS) $(GTK3LIB) \
 	    $(LIBDIR)/$(GTK3LIB_SONAME)
@@ -277,14 +257,6 @@
 	$(call quiet_cmd,LINK) $(CFLAGS) $(LDFLAGS) $(BIN_LDFLAGS) \
 	    -rdynamic -o $@ $(OBJS) $(LIBXNVCTRL) $(LIBS)
 
-$(eval $(call DEBUG_INFO_RULES, $(GTK2LIB)))
-$(GTK2LIB).unstripped: $(GTK2_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) $(VERSION_MK)
-	$(call quiet_cmd,LINK) -shared $(CFLAGS) $(LDFLAGS) $(BIN_LDFLAGS) \
-	    $(LIBXNVCTRL) $(LIBS) $(GTK2_LIBS) \
-	    -Wl,--unresolved-symbols=ignore-all -o $@ \
-	    -Wl,-soname -Wl,$(GTK2LIB_SONAME) \
-	    $(GTK2_OBJS) $(IMAGE_OBJS) $(XCP_OBJS)
-
 ifdef BUILD_GTK3LIB
 $(eval $(call DEBUG_INFO_RULES, $(GTK3LIB)))
 $(GTK3LIB).unstripped: $(GTK3_OBJS) $(XCP_OBJS) $(IMAGE_OBJS) $(VERSION_MK)
@@ -302,12 +274,9 @@
 clean clobber:
 	rm -rf $(NVIDIA_SETTINGS) *~ \
 		$(OUTPUTDIR)/*.o $(OUTPUTDIR)/*.d \
-		$(GTK2LIB) $(GTK3LIB) $(GTK2LIB_DIR) $(GTK3LIB_DIR) \
+		$(GTK3LIB) $(GTK3LIB_DIR) \
 		$(IMAGE_HEADERS) $(LIBXNVCTRL)
 
-$(foreach src,$(GTK_SRC), \
-    $(eval $(call DEFINE_OBJECT_RULE_WITH_DIR,TARGET,$(src),$(GTK2LIB_DIR))))
-
 ifdef BUILD_GTK3LIB
 $(foreach src,$(GTK_SRC), \
     $(eval $(call DEFINE_OBJECT_RULE_WITH_DIR,TARGET,$(src),$(GTK3LIB_DIR))))
@@ -320,7 +289,6 @@
 # Build $(IMAGE_HEADERS)
 $(foreach png,$(IMAGE_FILES), \
   $(eval $(call BINARY_DATA_HEADER_RULE,$(png))))
-$(GTK2_OBJS): $(IMAGE_HEADERS)
 ifdef BUILD_GTK3LIB
   $(GTK3_OBJS): $(IMAGE_HEADERS)
 endif
