--- a/nvidia-settings-@PV@/src/libXNVCtrl/xnvctrl.mk
+++ b/nvidia-settings-@PV@/src/libXNVCtrl/xnvctrl.mk
@@ -36,8 +36,8 @@
 
 XNVCTRL_CFLAGS ?=
 
-
-LIBXNVCTRL = $(OUTPUTDIR)/libXNVCtrl.a
+LIBXNVCTRLNAME = libXNVCtrl.so
+LIBXNVCTRL = $(OUTPUTDIR)/$(LIBXNVCTRLNAME)
 
 LIBXNVCTRL_SRC = $(XNVCTRL_DIR)/NVCtrl.c
 
@@ -46,4 +46,8 @@
 $(eval $(call DEFINE_OBJECT_RULE,TARGET,$(LIBXNVCTRL_SRC)))
 
 $(LIBXNVCTRL) : $(LIBXNVCTRL_OBJ)
-	$(call quiet_cmd,AR) ru $@ $(LIBXNVCTRL_OBJ)
+	$(call quiet_cmd,RM) $@ $@.*
+	$(CC) -shared -Wl,-soname=$@.0 -o $@.$(NVIDIA_VERSION) $(LDFLAGS) $^ -lXext -lX11
+	ln -s $@.$(NVIDIA_VERSION) $@.0
+	ln -s $@.0 $@
+
--- a/nvidia-settings-@PV@/src/Makefile
+++ b/nvidia-settings-@PV@/src/Makefile
@@ -252,6 +252,11 @@
 	$(MKDIR) $(BINDIR)
 	$(INSTALL) $(INSTALL_BIN_ARGS) $< $(BINDIR)/$(notdir $<)
 
+	$(MKDIR) $(LIBDIR)
+	$(INSTALL) $(INSTALL_LIB_ARGS) $(LIBXNVCTRLNAME).$(NVIDIA_VERSION) $(LIBDIR)
+	ln -s $(LIBXNVCTRLNAME).$(NVIDIA_VERSION) $(LIBDIR)/$(LIBXNVCTRLNAME).0
+	ln -s $(LIBXNVCTRLNAME).$(NVIDIA_VERSION) $(LIBDIR)/$(LIBXNVCTRLNAME)
+
 $(eval $(call DEBUG_INFO_RULES, $(NVIDIA_SETTINGS)))
 $(NVIDIA_SETTINGS).unstripped: $(OBJS) $(LIBXNVCTRL)
 	$(call quiet_cmd,LINK) $(CFLAGS) $(LDFLAGS) $(BIN_LDFLAGS) \
