--- a/nvidia-settings-@PV@/doc/nvidia-settings.desktop
+++ b/nvidia-settings-@PV@/doc/nvidia-settings.desktop
@@ -1,11 +1,11 @@
 [Desktop Entry]
 Type=Application
-Encoding=UTF-8
 Name=NVIDIA X Server Settings
 Comment=Configure NVIDIA X Server Settings
-Exec=__UTILS_PATH__/nvidia-settings
-Icon=__PIXMAP_PATH__/nvidia-settings.png
-Categories=__NVIDIA_SETTINGS_DESKTOP_CATEGORIES__
+Exec=nvidia-settings
+Icon=nvidia-settings
+Categories=System;Settings
+Keywords=nvidia;vulkan;opengl;driver
 
 # Translations provided by Sun Microsystems
 Name[de]=NVIDIA X-Server-Einstellungen
