--- a/SDK/CMake/FindOptiX.cmake
+++ b/SDK/CMake/FindOptiX.cmake
@@ -29,7 +29,7 @@
 # Locate the OptiX distribution.  Search relative to the SDK first, then look in the system.
 
 # Our initial guess will be within the SDK.
-set(OptiX_INSTALL_DIR "${CMAKE_SOURCE_DIR}/../" CACHE PATH "Path to OptiX installed location.")
+set(OptiX_INSTALL_DIR "${CMAKE_SOURCE_DIR}/../../" CACHE PATH "Path to OptiX installed location.")
 
 # The distribution contains only 64 bit libraries.  Error when we have been mis-configured.
 if(NOT CMAKE_SIZEOF_VOID_P EQUAL 8)
