--- a/SDK/optixCallablePrograms/CMakeLists.txt
+++ b/SDK/optixCallablePrograms/CMakeLists.txt
@@ -34,6 +34,7 @@
   )
 
 target_link_libraries( ${target_name}
+  glad
   ${CUDA_LIBRARIES}
   )
 
--- a/SDK/optixCutouts/CMakeLists.txt
+++ b/SDK/optixCutouts/CMakeLists.txt
@@ -34,6 +34,7 @@
   )
 
 target_link_libraries( ${target_name}
+  glad
   ${CUDA_LIBRARIES}
   )
 
--- a/SDK/optixDynamicMaterials/CMakeLists.txt
+++ b/SDK/optixDynamicMaterials/CMakeLists.txt
@@ -34,6 +34,7 @@
   )
 
 target_link_libraries( ${target_name}
+  glad
   ${CUDA_LIBRARIES}
   )
 
--- a/SDK/optixMeshViewer/CMakeLists.txt
+++ b/SDK/optixMeshViewer/CMakeLists.txt
@@ -34,6 +34,7 @@
   )
 
 target_link_libraries( ${target_name}
+  glad
   ${CUDA_LIBRARIES}
   )
 
--- a/SDK/optixMultiGPU/CMakeLists.txt
+++ b/SDK/optixMultiGPU/CMakeLists.txt
@@ -34,6 +34,7 @@
   )
 
 target_link_libraries( ${target_name}
+  glad
   ${CUDA_LIBRARIES}
   )
 
--- a/SDK/optixNVLink/CMakeLists.txt
+++ b/SDK/optixNVLink/CMakeLists.txt
@@ -34,6 +34,7 @@
   )
 
 target_link_libraries( ${target_name}
+  glad
   ${CUDA_LIBRARIES}
   )
 
--- a/SDK/optixPathTracer/CMakeLists.txt
+++ b/SDK/optixPathTracer/CMakeLists.txt
@@ -34,6 +34,7 @@
   )
 
 target_link_libraries( ${target_name}
+  glad
   ${CUDA_LIBRARIES}
   )
 
--- a/SDK/optixSimpleMotionBlur/CMakeLists.txt
+++ b/SDK/optixSimpleMotionBlurCMakeLists.txt
@@ -34,6 +34,7 @@
   )
 
 target_link_libraries( ${target_name}
+  glad
   ${CUDA_LIBRARIES}
   )
 
--- a/SDK/optixWhitted/CMakeLists.txt
+++ b/SDK/optixWhitted/CMakeLists.txt
@@ -34,6 +34,7 @@
   )
 
 target_link_libraries( ${target_name}
+  glad
   ${CUDA_LIBRARIES}
   )
 
